Fix brand count calculation and REST API endpoint

nerrad opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes incorrect brand count calculation and REST API endpoint behavior. The product_brand taxonomy was using WordPress's default _update_post_term_count callback instead of WooCommerce's custom _wc_term_recount function, and was not included in the wc_change_term_counts filter. This caused brand counts to return 0 instead of the correct count.

**Key Changes:**
- Changed product_brand taxonomy to use _wc_term_recount callback instead of _update_post_term_count
- Added product_brand to the woocommerce_change_term_counts filter to respect product visibility settings
- Enhanced brand tests with comprehensive coverage for count calculation, caching, and visibility
- Added complete REST API test suite for the /wp-json/wc/v3/products/brands endpoint
- Updated woo-phpunit rule with improved test execution documentation

Closes #60121.

### Screenshots or screen recordings:

N/A

### How to test the changes in this Pull Request:

1. Create a brand and assign it to a product
2. Check the brand count in the REST API: GET /wp-json/wc/v3/products/brands
3. Verify the count shows correctly (should be 1, not 0)
4. Set the product to out of stock and enable "Hide out of stock items" setting
5. Verify the brand count becomes 0 when the product is hidden
6. Check the brand count in the REST API: GET /wp-json/wc/v3/products/brands/{brand_id} request.
7. Verify the count shows correctly (should be 1, not 0 - because on brand specific requests, the visibility query is ignored).
8. Run the brand tests: pnpm run test:php:env tests/php/includes/class-wc-brands-test.php --verbose
9. Run the REST API tests: pnpm run test:php:env tests/php/includes/rest-api/Controllers/Version3/class-wc-rest-product-brands-controller-test.php --verbose

### Testing that has already taken place:

- All brand count tests pass (7 tests, 20 assertions)
- REST API tests pass (5 tests, 29 assertions)
- Verified both admin and frontend context behaviors
- Tested product visibility settings and dynamic count updates
- Confirmed fix resolves the issue described in #60121

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.
-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

Fixed incorrect count for brands through the /wp-json/wc/v3/products/brands endpoint. The product_brand taxonomy was using WordPress's default _update_post_term_count callback instead of WooCommerce's custom _wc_term_recount function, and was not included in the wc_change_term_counts filter. This caused brand counts to return 0 instead of the correct count.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

</details>

nerrad committed:
Fix incorrect count for brands through the /wp-json/wc/v3/products/brands endpoint

- Changed product_brand taxonomy to use _wc_term_recount instead of _update_post_term_count
- Added product_brand to the valid taxonomies in wc_change_term_counts filter
- This ensures brand counts properly account for WooCommerce product visibility logic

nerrad committed:
Fix brand count test to respect product visibility settings

- Use go_to() to simulate frontend context instead of admin context
- Use get_terms() instead of get_term() to trigger wc_change_term_counts filter
- Test now properly validates that brand counts respect 'hide out of stock items' setting

nerrad committed:
Add comprehensive brand count tests for admin and frontend contexts

- Add test_brand_count_respects_product_visibility() that uses go_to() and get_terms() to simulate frontend behavior
- Add test_brand_count_ignores_product_visibility_in_admin_context() that uses get_term() to simulate admin behavior
- Tests demonstrate different behaviors: frontend respects out-of-stock settings, admin ignores them
- Both tests pass and provide complete coverage of the context-dependent behavior

nerrad committed:
Add REST API brands controller test file

- Add comprehensive test suite for WC_REST_Product_Brands_Controller
- Tests cover brand listing, count accuracy, product visibility respect
- Includes test for brand counts respecting out-of-stock product settings
- All 5 tests pass with 29 assertions
- Provides end-to-end testing of brands API functionality

nerrad committed:
Remove admin vs frontend context documentation from woo-phpunit rule

The context-dependent behavior documentation was removed as it was found to be inaccurate:
- is_admin() returns false in test environment, not true
- The real issue is function choice (get_terms vs get_term), not context
- Simplified rule to focus on core testing practices

nerrad commented on .cursor/rules/woo-phpunit.mdc:
```diff
@@ -19,4 +20,67 @@ pnpm run test:php:env {relative_path} --verbose
 
 And the command must be run in the `plugins/woocommerce` directory.
 
+## Test Execution Best Practices
+
+### Command Structure
+
```
Improvements to the rule for PHPUnit tests seemed to help with a few additional things that popped up using the agent.

nerrad commented on plugins/woocommerce/includes/wc-term-functions.php:
```diff
@@ -323,12 +323,12 @@ function wc_reorder_terms( $the_term, $next_id, $taxonomy, $index = 0, $terms =
 		// the nextid of our term to order, lets move our term here.
 		if ( null !== $next_id && $term_id === $next_id ) {
-			$index++;
+			++$index;
```
Autoformatting did these changes. Shouldn't matter (but could if this was in the context where the value was being output).

nerrad committed:
Remove duplicate product_brand filter from WC_Brands class

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

nerrad commented on plugins/woocommerce/includes/class-wc-brands.php:
```diff
@@ -89,18 +88,6 @@ public function register_hooks() {
 		add_filter( 'hooked_block_core/post-terms', array( $this, 'configure_product_brand_block' ), 10, 5 );
 	}
 
-	/**
-	 * Add product_brand to the taxonomies overridden for the original term count.
-	 *
-	 * @param array $taxonomies List of taxonomies.
-	 *
-	 * @return array
-	 */
-	public function add_brands_to_terms( $taxonomies ) {
-		$taxonomies[] = 'product_brand';
-		return $taxonomies;
-	}
-
```
This is removed (along with the filter registering this callback) because I shifted this to the `apply_filters( 'woocommerce_change_term_counts' )` hook execution in `wc_change_term_counts` function. This eliminates some unnecessary lines of code and simplifies things. Overtime whenever touching the brands code we should try and move more of it natively in core (unfortunately, it was implemented as a mostly a copy of the original plugin code).

sunyatasattva replied:
> Overtime whenever touching the brands code we should try and move more of it natively in core 

💯  agreed!

nerrad committed:
Delete plugins/woocommerce/changelog/fix-wooplug-5231-incorrect-count-for-brands-through-the-wp

Remove superfluous changelog added by AI.

nerrad reviewed (commented):
I also added some tests to improve coverage of brands behaviour.

nerrad committed:
update brands rest controller test to remove unnecessary setup

joshuatf commented on .cursor/rules/woo-phpunit.mdc:
```diff
@@ -19,4 +20,67 @@ pnpm run test:php:env {relative_path} --verbose
+If you get "Class could not be found" errors:
+
+-   Use the `--filter` approach instead of direct file paths
+-   Ensure test classes have `public` setUp() and tearDown() methods (not `protected`)
```
Seems like a bit of duplication between this and "2. Test Method Visibility" below.

Have you found this helps with cursor rules?

nerrad replied:
Hmm... I think these were both lingering artifacts from me reviewing what AI added - the existing file worked well, but I'll remove the duplications for now. I have noticed that it does seem re-emphasizing things helps but it also increases the token count for context, so I'll remove.

joshuatf commented on .cursor/rules/woo-phpunit.mdc:
```diff
@@ -19,4 +20,67 @@ pnpm run test:php:env {relative_path} --verbose
+
+-   Use the `--filter` approach instead of direct file paths
+-   Ensure test classes have `public` setUp() and tearDown() methods (not `protected`)
+-   Add `declare( strict_types = 1 );` at the top of test files
```
Same comment here with strict types.

joshuatf commented on plugins/woocommerce/tests/php/includes/rest-api/Controllers/Version3/class-wc-rest-product-brands-controller-test.php:
```diff
@@ -0,0 +1,211 @@
+	/**
+	 * Test that brand counts are updated when products are added/removed.
+	 */
+	public function test_brand_counts_update_when_products_change() {
```
Should we also add a test to check when brands are deleted?

nerrad replied:
Ya, doesn't hurt. I _think_ it's probably covered by WP core taxonomy tests, but with the amount of filtering Woo has in place, I think it's good to still have it to catch potential regressions in the future.

nerrad replied:
See fc1f0c1eb80606d49a26394ff3f3545b17d63a97

joshuatf reviewed (commented):
Thanks for the changes, @nerrad!  This is mostly testing well- just left a couple of comments and a couple questions below around expectations.

At this PR size, it might be worth splitting into multiple PRs for future PRs, especially when we have an easy way to separate and test functionality (testing rules and brands endpoint).

✅ Cursor uses new rules to help run tests and determines how to correctly run a single file
✅ Brands endpoint returns correct count of brands when adding or removing brands
> Verify the count shows correctly (should be 1, not 0)

❓ Is this meant to be under the `/wp-json/wc/v3/products/brands/{id}` endpoint?  I don't believe the count will change on the `/wp-json/wc/v3/products/brands` endpoint
> Verify the brand count becomes 0 when the product is hidden

🛑 Assuming the above is under the individual item endpoint, I set the product out of stock hidden settings under `/wp-admin/admin.php?page=wc-settings&tab=products&section=inventory`, and still got a `count` of `1` when the brand was assigned to that product, despite it being set to "Out of Stock" and not visible in the frontend.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -13,6 +13,16 @@
  */
 class WC_Brands_Test extends WC_Unit_Test_Case {
+
+	/**
+	 * Tear down test data.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+
+		// Clear term cache to prevent interference between tests.
```
NIT: These comments don't add much context to the code. I've noticed that we use a similar docblock across the codebase, so we could keep that for consistency. Though I'm not a fan of comments that only add noise to the code.

nerrad replied:
Fair, but I also don't think they, hurt. Especially in the age of AI, where it will often use comments to get better context.

sunyatasattva replied:
I disagree, but as I said it was a nit and I don't have a strong opinion. I am not sure whether we are optimizing for AI or human readability. In both cases, I still think that a comment such as “Get the brand term” below, just before a call to `get_first_brand_term` only adds noise for both AI and humans. Not costly for AI, slightly costly for humans.

This comment here at least adds some context as to why the term cache is being cleared (though it's usually obvious on functions such as `tearDown`).

nerrad replied:
Ya, I get it. I'll watch in future PRs for overly superfluous comments.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +84,126 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+	public function test_brand_count_calculation_and_caching() {
+		$data = $this->setup_brand_test_data();
+
+		// Get the brand term.
```
I'm not going to repeat it for every time, but most of these comments are unnecessary. Code is pretty self-documenting and comments also repeat what the assertion text is already saying.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +84,126 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		$data    = $this->setup_brand_test_data();
+		$product = $data['product'];
+
+		// Enable hide out of stock setting FIRST.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'yes' );
+
+		// THEN set product to out of stock (hook will fire with correct setting).
+		$product->set_stock_status( 'outofstock' );
+		$product->save();
```
The comments make me think that if the order is inverted this won't work. Is that so? Wouldn't that be problematic?

nerrad replied:
Which order, setting the option, or `save()` after `set_stock_status`? We're setting up the environment to test expectations. Expectations being:

- If you have the site configured to hide out of stock items...
- ...and the product has its stock status to be out of stock...
- ...and the product is saved.

sunyatasattva replied:
I meant that the comment said “hook will fire with correct settings”, underscoring the order of `FIRST` and `THEN`: does it mean that if you set a product out of stock FIRST, and THEN set the option something wouldn't work?

The comments seem to read as if it is the case, but it would be odd if it did.

nerrad replied:
Oh I get ya, the emphasis on `THEN` seems to imply specific order. It doesn't matter, but I get your point.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +84,126 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+				'hide_empty' => false,
+			)
+		);
+		$brand_term  = $brand_terms[0];
```
Tests repeat many times this block. I'd just create a helper function called `get_first_brand_term`. This will reduce noise across all the tests.

nerrad replied:
Good call.

nerrad replied:
See ef1da099

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +84,126 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		$brand_term = get_term( $data['brand_with_products']['term_id'], 'product_brand' );
+		$this->assertEquals( 1, $brand_term->count, 'Brand should initially have 1 product' );
+
+		// Remove the product from the brand.
+		wp_set_object_terms( $product->get_id(), array(), 'product_brand' );
```
As a positive example of a good comment, I'd keep this one, because by reading the code alone I needed an extra mental step to understand it.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +84,126 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		wp_set_object_terms( $product->get_id(), array( $data['brand_with_products']['term_id'] ), 'product_brand' );
+
+		// Count should be updated back to 1.
+		$brand_term_final = get_term( $data['brand_with_products']['term_id'], 'product_brand' );
```
I'd also refactor this `$data['brand_with_products']['term_id']` into a shorter variable. I'd even go one step further and say I'd make the whole function into its own helper since it's duplicated so many times. Calling it `get_test_brand_term` or something will greatly reduce noise in the code.

nerrad replied:
The suggestion you made earlier made sense because it impacted a larger block of repeated code. I don't really see the value of an additional abstraction over `get_term`.

sunyatasattva replied:
I think that the value here is that the operation performed is the same 3 times in a row and it would help underscoring that. At the very least, I would refactor `$data['brand_with_products']['term_id']` in a shorter `$term_id`. There is no reason to keep it full, IMO, as I'm forced to read the whole string to make sure that I'm not missing something and that it is indeed the same data that's being fetched.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/rest-api/Controllers/Version3/class-wc-rest-product-brands-controller-test.php:
Ok I might be saying something really stupid here, but hear me out: do we _really_ need this test file?

You are testing the functions in the previous file, and the Brands controller just extends the `WC_REST_Product_Categories_Controller`, so it should be mostly the responsibility of the latter to test anything.

I understand that you are testing specifically the counts, but many of these tests seem somewhat unnecessary to me: they are using the functions already tested above and basically just testing that REST functionality works as intended.

Again, this might be a stupid remark and these tests are pretty cheap, but that was my feeling going through the whole thing.

nerrad replied:
Unless I missed something, there is no terms controller or categories controller testing counts within the REST request context. Were there specific tests in other files you had in mind that already cover this? AFAICT, there were no tests that would have caught the reported bug, so I added these to make sure no regressions surfaced in the future.

These tests (and more 😬) were initially generated by AI, but I removed several superfluous ones as well as reviewed/modified the existing. The latter three tests deal with counts in specific conditions (visibility on product, a request with include parameter, and that the counts update when products_change), which I think are important for catching any potential regressions. 

In particular, the visibility test is interesting because according to the behavior that exists in the code on **REST Requests** to this endpoint (not changed in this PR), the count of products on a brand will not include products that are out of stock. Again, [_current_ behavior](https://github.com/woocommerce/woocommerce/blob/62f844a3f3b63946d4bcd0d369362966e63ff0b3/plugins/woocommerce/includes/wc-term-functions.php#L561-L563), but it could be problematic in some contexts where the REST API is being used for UIs where the visibility rules should be irrelevant (editing a product, viewing a list of brands etc). Contrast that with the `WC_Brands_Test::test_brand_count_ignores_product_visibility_in_admin_context` test, which demonstrates that in the _admin_ (web) context, the visibility rules _are_ ignored.

sunyatasattva replied:
> Were there specific tests in other files you had in mind that already cover this?

I don't think there are. My point was more that the test about reacting correctly within the REST context are perhaps responsibility of a `WC_REST_Product_Categories_Controller` test, and this bug would be covered by the other test cases you added above _and_ some tests we could add to the `WC_REST_Product_Categories_Controller`.

It's basically a matter of responsibility and where test lives and avoid duplication of already tested functionality.

nerrad replied:
Ahh, gotcha. I get your point. Makes sense to move.

sunyatasattva reviewed (commented):
I agree with @joshuatf 's comments above. Other than that, the implementation is pretty straight-forward and clean, I love it. I left a few comments mostly on the tests.

nerrad committed:
Update doc block for method to include Brands

nerrad committed:
Add test for deleting product a brand is attached to.

nerrad committed:
Simplify the tests with helper for getting first brand term.

nerrad reviewed (commented):
Thanks, Lucio for the review. I've made a few changes and left comments about the others.

nerrad commented:
Thanks for your review too @joshuatf!

> At this PR size, it might be worth splitting into multiple PRs for future PRs, especially when we have an easy way to separate and test functionality (testing rules and brands endpoint).

Are you suggesting splitting the tests files out from the changes that were implemented? If so, I disagree. I think any additional tests should be in the same PR with changes being tested. They validate the changes and also set expectations correctly for what the changes _should_ be doing. Important in this case too because the bug was a regression in brand's behavior (as a part of its move into Woo Core).

If you're referring to the Cursor rule change being bundled in this PR, then yes, that is something I could have pulled out to a separate PR. I thought the changes were minimal and this wasn't really a larger PR to begin with so included it. I'm happy to not do that in future PRs :) 

> ❓ Is this meant to be under the /wp-json/wc/v3/products/brands/{id} endpoint? I don't believe the count will change on the /wp-json/wc/v3/products/brands endpoint

So related to the above and your subsequent comment, `wp-json/wc/v3/products/brands/` (not specific brand) is returning count as expected for how out of stock items is configured (and matches the controller tests). However, requesting a _specific_ brand (`wp-json/wc-v3/products/brands/27`) always shows the count of products on it, regardless of the out-of-stock setting. I'm not sure if that is how it's supposed to be, but that's the current behavior (that existed before this branch). So current testing steps in the description are correct (and testing as expected for me and the unit tests). However, I could also add unit (integration) tests for the _specific brand_ endpoint to reflect that counts do not change on that endpoint with the relevant out of stock configuration.

nerrad committed:
Add test for visibility affected count on  specific brand REST request.

nerrad commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -7,12 +7,20 @@
  */
 class WC_Brands_Test extends WC_Unit_Test_Case {
+
+	/**
+	 * Tear down test data.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+
+		// Clear term cache to prevent interference between tests.
+		clean_term_cache( array(), 'product_brand' );
+	}
```
- current screen is reset via inheritance of WP_UnitTestCase_Base::tearDown.
- clean_term_cache with empty array cleans the entire taxonomy cache (the default is to clean the taxonomy cache).

So state is, in fact, reset.

nerrad commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +82,105 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		$this->assertEquals( '1', $cached_count, 'Brand count should be cached in term meta' );
+	}
+
+	/**
+	 * Test that brand counts respect product visibility settings.
+	 */
+	public function test_brand_count_respects_product_visibility() {
+		$data    = $this->setup_brand_test_data();
+		$product = $data['product'];
+
+		// Enable hide out of stock setting FIRST.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'yes' );
+
+		// THEN set product to out of stock (hook will fire with correct setting).
+		$product->set_stock_status( 'outofstock' );
+		$product->save();
+
+		// Get the brand term.
+		$brand_term = $this->get_first_brand_term( array( $data['brand_with_products']['term_id'] ) );
+
+		// Test that the count is 0 when product is out of stock and hidden.
+		$this->assertEquals( 0, $brand_term->count, 'Brand count should be 0 when product is out of stock and hidden' );
+
+		// Test that the count is cached correctly.
+		$cached_count = get_term_meta( $brand_term->term_id, 'product_count_product_brand', true );
+		$this->assertEquals( '0', $cached_count, 'Brand count should be cached as 0 when product is out of stock' );
+
+		// Reset the setting.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'no' );
+	}
```
This would only be needed if there is a potential exception thrown. I don't think that's a possibility here, and a fatal error will break tests anyways.

nerrad commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +82,105 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		$this->assertEquals( 1, $brand_term_final->count, 'Brand should have 1 product after re-adding' );
+	}
+
+	/**
+	 * Test that brand counts ignore product visibility in admin context.
+	 */
+	public function test_brand_count_ignores_product_visibility_in_admin_context() {
+		$data    = $this->setup_brand_test_data();
+		$product = $data['product'];
+
+		// Enable hide out of stock setting.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'yes' );
+
+		// Set product to out of stock.
+		$product->set_stock_status( 'outofstock' );
+		$product->save();
+
+		// Set admin context.
+		set_current_screen( 'edit-post' );
+
+		// Get the brand term using helper.
+		$brand_term = $this->get_first_brand_term( array( $data['brand_with_products']['term_id'] ) );
+
+		// Test that the count is 1 (ignores out of stock setting in admin context).
+		$this->assertEquals( 1, $brand_term->count, 'Brand count should be 1 in admin context, ignoring out of stock setting' );
+
+		// Reset the setting.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'no' );
+	}
```
Reset of the current screen happens via inheritance on `tearDown`.

sunyatasattva commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -74,6 +82,105 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
+		// Get the brand term.
+		$brand_term = $this->get_first_brand_term( array( $data['brand_with_products']['term_id'] ) );
+
+		// Test that the count is 0 when product is out of stock and hidden.
+		$this->assertEquals( 0, $brand_term->count, 'Brand count should be 0 when product is out of stock and hidden' );
```
Just as a further clarification about the comments: look at this comment. It literally says exactly what the assertion string below says. As a human, I'm naturally reading the comment as I think it's going to add something valuable, and I'll fill up my “human context” 🙃

sunyatasattva reviewed (approved):
Approving the PR. All the comments I made are not mandatory. See if you like any of the suggestions, otherwise I'm happy for this to be merged. Thanks for working on it!

nerrad committed:
Move tests to category controller.

The brands controller extends the category controller so this ensures tests live in the appropriate place.

As a part of this refactor I also improved some of the test architecture and removed superfluous term_cache clears (WP scaffolding already clears caches on tearDown).

nerrad commented on plugins/woocommerce/tests/php/includes/class-wc-brands-test.php:
```diff
@@ -96,4 +203,22 @@ private function setup_brand_test_data() {
 		);
 	}
+
+	/**
+	 * Helper method to get the first brand term.
+	 *
+	 * @param array $term_ids Array of brand term IDs to include.
+	 * @return WP_Term The first brand term.
+	 */
+	private function get_first_brand_term( $term_ids = array() ) {
+		$args = array(
+			'taxonomy'   => 'product_brand',
+			'hide_empty' => false,
+		);
+		if ( ! empty( $term_ids ) ) {
+			$args['include'] = $term_ids;
+		}
+		$brand_terms = get_terms( $args );
+		return $brand_terms[0];
+	}
```
Good observation (and generally valid), but in this case since this is in a test suite, any issues with that will immediately surface in failing tests.
